src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -319,7 +326,10 @@ def _slice_template(self) -> List[RawFileSlice]:
                                 f"\0{prefix}{unique_alternate_id} {open_} "
                                 f"{trimmed_content} {close_}"
                             )
-                if block_type == "block_start" and trimmed_content.split()[0] == "set":
+                if block_type == "block_start" and trimmed_content.split()[0] in (
+                    "macro",
+                    "set",
+                ):
```

Comment:
To fix this bug, we need to treat macro definitions the same as `{% set %}` blocks.